src/lib/utils/MathLib.ts

```diff
@@ -9,19 +9,46 @@ import { Vector2 } from 'three';
  * @param distance The minimum required distance from the point to the line
  * @returns the closest point on the line or null
  */
-export function snapPointToLine(point: Vector2, point1_line: Vector2, point2_line: Vector2, distance: number) {
-    const a = point1_line.y - point2_line.y;
-    const b = point2_line.x - point1_line.x;
-    const c = point1_line.x * point2_line.y - point2_line.x * point1_line.y;
-    const actual_distance = Math.abs(a * point.x + b * point.y + c) / Math.sqrt(a * a + b * b);
+export function snapPointToLine(
+  point: Vector2,
+  point1_line: Vector2,
+  point2_line: Vector2,
+  distance: number
+) {
+  const a = point1_line.y - point2_line.y;
+  const b = point2_line.x - point1_line.x;
+  const c = point1_line.x * point2_line.y - point2_line.x * point1_line.y;
+  const actual_distance = Math.abs(a * point.x + b * point.y + c) / Math.sqrt(a * a + b * b);
 
-    //Calculate the closest point from the line to point q
-    const x3 = (b * b * point.x - a * b * point.y - a * c) / (a * a + b * b);
-    const y3 = (a * a * point.y - a * b * point.x - b * c) / (a * a + b * b);
+  //Calculate the closest point from the line to point q
+  const x3 = (b * b * point.x - a * b * point.y - a * c) / (a * a + b * b);
+  const y3 = (a * a * point.y - a * b * point.x - b * c) / (a * a + b * b);
 
-    if (actual_distance < distance) {
-      return new Vector2(x3, y3)
-    }
+  if (actual_distance < distance) {
+    return new Vector2(x3, y3);
+  }
 
-    return null
-}
\ No newline at end of file
+  return null;
+}
+
+export function lineLineIntersection(A: Vector2, B: Vector2, C: Vector2, D: Vector2) {
```

Comment:
A jsDoc comment like the snapPointToLine would be great here

Reply:
I have added it